File: android/src/main/java/com/enw/plugins/nativehash/NativeHashPlugin.kt
package com.enw.plugins.nativehash

import java.security.MessageDigest
import android.util.Base64
import com.getcapacitor.PluginCall
import com.getcapacitor.Plugin
import com.getcapacitor.annotation.CapacitorPlugin
import com.getcapacitor.PluginMethod

@CapacitorPlugin(name = "NativeHash")
class NativeHashPlugin : Plugin() {
    @PluginMethod
    fun sha256(call: PluginCall) {
        val input = call.getString("text") ?: run {
            call.reject("Missing 'text'")
            return
        }

        val digest = MessageDigest.getInstance("SHA-256")
        val hash = digest.digest(input.toByteArray())
        val base64Hash = Base64.encodeToString(hash, Base64.NO_WRAP)

        val ret = JSObject().put("hash", base64Hash)
        call.resolve(ret)
    }
}

File: ios/Sources/NativeHashPlugin/NativeHashPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(NativeHashPlugin)
public class NativeHashPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "NativeHashPlugin"
    public let jsName = "NativeHash"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]

    @objc func sha256(_ call: CAPPluginCall) {
        guard let text = call.getString("text") else {
            call.reject("Missing 'text'")
            return
        }

        let data = Data(text.utf8)
        let digest = SHA256.hash(data: data)
        let hashData = Data(digest)
        let base64Hash = hashData.base64EncodedString()

        call.resolve([
            "hash": base64Hash
        ])
    }
}
